### PR TITLE
Upgrade Node and dependencies

### PR DESCRIPTION
**Pull request by elfenheart**
I have updated things to minimize deprecation notice, however there are still some I could not fix, such as Gulp and Vega.

I have also migrate npm to use yarn.

`yarn.lock` file is a generated file, so no need to review that.

### CODE REVIEW

File: cdap-ui/package.json [hunk 1]
```diff
@@ -33,34 +33,32 @@
...
-    "browser-sync": "2.18.8",
+    "bower": "1.8.2",
     "case-sensitive-paths-webpack-plugin": "2.1.1",
-    "concurrently": "3.1.0",
     "copy-webpack-plugin": "4.0.1",
     "css-loader": "0.28.7",
     "del": "2.2.0",
     "enzyme": "2.6.0",
-    "es6-promise": "3.0.2",
+    "es6-promise": "4.1.1",
```
Comment (ajainarayanan) [inline cdap-ui/package.json]: nit: Now that we run on node 8+ do we need this polyfill? If yes can we see if we can try to re-use babel-polyfill? It already has an implementation of promise.

File: cdap-ui/pom.xml [hunk 2]
```diff
@@ -190,84 +190,68 @@
...
+                  <arguments>install --silent --frozen-lockfile</arguments>
                 </configuration>
               </execution>
+
               <execution>
                 <id>bower-install</id>
-                <phase>process-resources</phase>
                 <goals>
-                  <goal>exec</goal>
+                  <goal>bower</goal>
```
Comment (ajainarayanan) [inline cdap-ui/pom.xml]: This will require bower to be installed globally? For someone who is just building sdk (backend folks) would this fail?

**Review by ajainarayanan — COMMENTED** [review]
Just minor comments. Otherwise awesome changes!! 🎉 🎆 🔥

File: cdap-ui/pom.xml (hunk 2 above)
Comment (elfenheart) [reply cdap-ui/pom.xml]: Seems to work without bower globally installed

**Review by ajainarayanan — APPROVED** [review]
🎆 🏆 LGTM

**Pull request merged**